src/identity/interaction/HtmlViewHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { cleanPreferences, getTypeWeight } from '../../storage/conversion/ConversionUtil';
+import { APPLICATION_JSON, TEXT_HTML } from '../../util/ContentTypes';
+import { MethodNotAllowedHttpError } from '../../util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import type { TemplateEngine } from '../../util/templates/TemplateEngine';
+import type { InteractionHandlerInput } from './InteractionHandler';
+import { InteractionHandler } from './InteractionHandler';
+import type { InteractionRoute } from './routing/InteractionRoute';
+
+/**
+ * Stores the HTML templates associated with specific InteractionRoutes.
+ * Template keys should be file paths to the templates,
+ * values should be the corresponding routes.
+ *
+ * Will only handle GET operations for which there is a matching template if HTML is more preferred than JSON.
+ *
+ * Templates will receive the parameter `idpIndex` in their context pointing to the root index URL of the IDP API.
+ */
+export class HtmlViewHandler extends InteractionHandler {
+  private readonly idpIndex: string;
+  private readonly templateEngine: TemplateEngine;
+  private readonly templates: Record<string, string>;
+
+  public constructor(index: InteractionRoute, templateEngine: TemplateEngine,
+    templates: Record<string, InteractionRoute>) {
+    super();
+    this.idpIndex = index.getPath();
+    this.templateEngine = templateEngine;
+    this.templates = Object.fromEntries(
+      Object.entries(templates).map(([ template, route ]): [ string, string ] => [ route.getPath(), template ]),
```

Comment:
Perhaps https://lodash.com/docs/#mapValues

Reply:
Also switching key and value here so won't work.